squire_sdk/src/actor.rs

```diff
@@ -97,6 +97,8 @@ pub struct Scheduler<A: ActorState> {
     recv: SelectAll<ActorStream<A>>,
     queue: FuturesUnordered<Pin<Box<dyn 'static + Send + Future<Output = A::Message>>>>,
     tasks: FuturesUnordered<Pin<Box<dyn 'static + Send + Future<Output = ()>>>>,
+    should_terminate: bool,
+    sender: UnboundedSender<A::Message>,
```

Comment:
I missed this during the peer programming. We might not want a regular sender here. Instead, we might want a downgraded sender.

Imagine we drop all clients to a given actor such that the only way to get a new client is from the scheduler. This means that the actor can't (currently) receive any messages from its primary channel and could be considered dead. However, the channel is not considered closed because the scheduler holds a sender. This could impede otherwise desired cleanup.

I don't think there needs to be a change from this PR, but I wanted to note this for future efforts.

Reply:
I will downgrade this along with the tests I write on sunday.